DQM/SiPixelPhase1Summary/src/SiPixelPhase1Summary.cc

```diff
@@ -283,6 +383,113 @@ void SiPixelPhase1Summary::fillSummaries(DQMStore::IBooker& iBooker, DQMStore::I
     deadROCSummary->setBinContent(xBin, yBin, nROCs / nRocsPerTrend[i]);
   }
 
+  //Fill additional residuals plots
+  //PXBarrel
+  for (std::string layer : {"1", "2", "3", "4"}) {
+    MonitorElement* me_x =
+        iGetter.get("PixelPhase1/Tracks/PXBarrel/residual_x_per_SignedModule_per_SignedLadder_PXLayer_" + layer);
+    MonitorElement* me_y =
+        iGetter.get("PixelPhase1/Tracks/PXBarrel/residual_y_per_SignedModule_per_SignedLadder_PXLayer_" + layer);
+
+    bool isLayer_4 = layer == "4";  //Order of inner and outer ladders is reversed for Layer 4
+
+    for (int i = 1; i <= me_x->getNbinsY(); i++) {
+      if (i == (me_x->getNbinsY() / 2 + 1))
+        continue;  //Middle bin of y axis is empty
+
+      if (i <= me_x->getNbinsY() / 2) {
+        if ((i % 2 == 0 && !isLayer_4) || (i % 2 == 1 && isLayer_4)) {
```

Comment:
@dmeuser I guess it will be very difficult for someone reading this code in 5 years to understand why we do this here. Can an ancillary variable be added here like it is done at https://github.com/cms-sw/cmssw/blob/master/DQM/SiPixelPhase1Track/plugins/SiPixelPhase1TrackClusters.cc#L181-L184 to make it clear the ladder order in BPix L4 is swapped?